src/docs/epics/77-maven-due-diligence/index.adoc

```diff
@@ -0,0 +1,222 @@
+---
+# Do not change this
+issue: 77
+# This might change in the course of the work
+title: Maven Due Dilligence
+# This might be one of TBD
+status: Draft
+---
+:skip-front-matter:
+= Maven Due Diligence
+:icons: font
+:toc: left
+
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+[.lead]
+.https://github.com/support-and-care/maven-support-and-care/issues/77[Epic 77]
+****
+The major task of this epic is to _investigate the current state of the Maven build_ and to ensure that we can build all projects with the required JDK and Maven versions.
+We want to use tools like https://jqassistant.org[jQAssistant] to analyze the quality of Maven and perform further analysis like overall dependency checks.
+A precondition for successful analysis is that we can build all projects with the respective JDK and Maven versions.
+
+[[box:maven-source]]
+[IMPORTANT]
+.Maven Source
+====
+Most of the content in this document refers to the so-called _Maven Source_ (cf. https://maven.apache.org/scm.html[SCM section] of the Maven documentation).
+Whenever we use this term, we refer to the collection of Maven components provided with the described SCM structure.
+====
+****
+
+== Summary and resulting tasks
+
+We would like to ensure that we can build all Maven projects with the required JDK and Maven versions
+
+* As _reactor build_ (cf. https://maven.apache.org/guides/development/guide-building-maven.html[Building Maven]), but also
+* As _individual_ projects (e.g., to check for outdated dependencies).
+
+=== Current state
+
+The Maven build is in a rather good state, taking into account that there are 133 subprojects developed and maintained since about 20 years by only a small number of committers.
+The build is well-structured and uses a reactor build to build most projects in a single run.
+However, there are some problems with the build, mainly due to outdated Java versions and dependencies.
+The <<sec:detailed-build-investigation,detailed investigation>> (current state) shows the problems which prohibit a successful build of the Maven projects.
+
+=== Resulting tasks
+
+==== Quick wins
+
+We derived following tasks from the current state above.
+They make sure that we can at least execute the Maven build with the required JDK and Maven versions without running into basic build problems.
+
+* [ ] Check with the Maven committers if we could thin out the _reactor build_ and the overall <<box:maven-source>>.
+** [ ] Archive unnecessary components.
+** [ ] Drop them from the Maven Source.
+
+* [ ] Update the Java version in all projects to at least Java 8.
```

Comment:
For Maven 4, I mean Java 17 is the at least version

Reply:
It currently builds with Java 8, so I would leave this as the minimal version (at least for non core projects, since they should work with Maven 3 by my understanding). 
But we can sort this out with the Maven team.

Reply:
Updating does not make sense yet, there is a simple reason:

* Plugins will still target the 3.6.x API of Maven.
* Plugins using the 3.6.x API can run on Maven 3.6.x or later,
* Maven 4 can run Maven Plugins built for 3.6.x or later

So, if we raised the minimum Java version for other projects, like plugins, they would become Maven 4-only and thus need a major version bump to v4.x.x. I do not think this is one of our goals. ;)

Reply:
BTW, some current modules (e.g. [Maven Resolver](https://github.com/apache/maven-resolver/blob/075483449e747e79dcbbe56491b48015633224ce/pom.xml#L120) already require JDK 21 explicitly.